test/jdk/javax/swing/border/Test4129681.java

```diff
@@ -21,39 +21,58 @@
  * questions.
  */
 
+import java.awt.event.ItemEvent;
+import javax.swing.BorderFactory;
+import javax.swing.Box;
+import javax.swing.JCheckBox;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+
 /*
  * @test
  * @bug 4129681
  * @summary Tests enabling/disabling of titled border's caption
- * @author Sergey Malenkov
- * @run applet/manual=yesno Test4129681.html
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual Test4129681
  */
 
-import java.awt.BorderLayout;
-import java.awt.event.ItemEvent;
-import java.awt.event.ItemListener;
-import javax.swing.BorderFactory;
-import javax.swing.JApplet;
-import javax.swing.JCheckBox;
-import javax.swing.JLabel;
+public class Test4129681 {
+    private static JLabel label;
 
-public class Test4129681 extends JApplet implements ItemListener {
-    private JLabel label;
+    public static void main(String[] args) throws Exception {
+        String testInstructions = """
+                Turn on the checkbox to disable the label.
+                The test passes if the title of the border
+                is disabled as well as the label.
+                """;
 
-    @Override
-    public void init() {
-        JCheckBox check = new JCheckBox("disable");
-        check.addItemListener(this);
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(testInstructions)
+                .rows(4)
+                .columns(25)
+                .splitUI(Test4129681::init)
+                .build()
+                .awaitAndCheck();
+    }
 
-        this.label = new JLabel("message");
-        this.label.setBorder(BorderFactory.createTitledBorder("label"));
-        this.label.setEnabled(!check.isSelected());
+    public static JComponent init() {
+        JCheckBox check = new JCheckBox("Enable/Disable");
+        check.addItemListener(event ->
+                label.setEnabled(ItemEvent.DESELECTED == event.getStateChange()));
 
-        add(BorderLayout.NORTH, check);
-        add(BorderLayout.CENTER, this.label);
-    }
+        label = new JLabel("message");
+        label.setBorder(BorderFactory.createTitledBorder("label"));
+        label.setEnabled(!check.isSelected());
 
-    public void itemStateChanged(ItemEvent event) {
-        this.label.setEnabled(ItemEvent.DESELECTED == event.getStateChange());
+        Box main = Box.createVerticalBox();
+        main.setBorder(BorderFactory.createEmptyBorder(8, 8, 8, 8));
+        main.add(Box.createVerticalGlue());
+        main.add(check);
+        main.add(Box.createVerticalStrut(4));
+        main.add(label);
+        main.add(Box.createVerticalGlue());
```

Comment:
A `JPanel` with `BorderLayout` works better here, you can fill the area with the `JLabel`:
```suggestion
        JPanel main = new JPanel(new BorderLayout());
        main.setBorder(BorderFactory.createEmptyBorder(8, 8, 8, 8));
        main.add(check, BorderLayout.NORTH);
        main.add(label, BorderLayout.CENTER);
```

Reply:
You didn't change to `JPanel` with `BorderLayout`. Whatever.